src/dali/plugin/input/rest/binance_com.py

```diff
@@ -0,0 +1,1292 @@
+
+        # Is this a plain buy or a conversion?
+        if trade.quote_asset in self.client.options[_LEGAL_MONEY]:  # Is this fiat? (options[_LEGAL_MONEY] is a list of all fiat ever traded on Binance.com)
+            fiat_out_no_fee: RP2Decimal = RP2Decimal(str(transaction[_COST]))
+            fiat_fee: RP2Decimal = RP2Decimal(crypto_fee)
+            spot_price: RP2Decimal = RP2Decimal(str(transaction[_PRICE]))
+
+            out_transaction_list.append(
+                OutTransaction(
+                    plugin=self.__BINANCE_COM,
+                    unique_id=transaction[_ID],
+                    raw_data=json.dumps(transaction),
+                    timestamp=self._rp2_timestamp_from_ms_epoch(transaction[_TIMESTAMP]),
+                    asset=out_asset,
+                    exchange=self.__BINANCE_COM,
+                    holder=self.account_holder,
+                    transaction_type=Keyword.SELL.value,
+                    spot_price=str(spot_price),
+                    crypto_out_no_fee=str(crypto_out_no_fee),
+                    crypto_fee=str(crypto_fee),
+                    crypto_out_with_fee=str(crypto_out_with_fee),
+                    fiat_out_no_fee=str(fiat_out_no_fee),
+                    fiat_fee=str(fiat_fee),
+                    fiat_ticker=trade.quote_asset,
+                    notes=(f"{notes + ';' if notes else ''} Fiat sell of {trade.base_asset} with {trade.quote_asset}."),
+                )
+            )
+
+        else:
+            # Binance does not report the value of transaction in fiat
+            out_transaction_list.append(
+                OutTransaction(
+                    plugin=self.__BINANCE_COM,
+                    unique_id=transaction[_ID],
+                    raw_data=json.dumps(transaction),
+                    timestamp=self._rp2_timestamp_from_ms_epoch(transaction[_TIMESTAMP]),
+                    asset=out_asset,
+                    exchange=self.__BINANCE_COM,
+                    holder=self.account_holder,
+                    transaction_type=Keyword.SELL.value,
+                    spot_price=Keyword.UNKNOWN.value,
+                    crypto_out_no_fee=str(crypto_out_no_fee),
+                    crypto_fee=str(crypto_fee),
+                    crypto_out_with_fee=str(crypto_out_with_fee),
+                    fiat_out_no_fee=None,
+                    fiat_fee=None,
+                    notes=(
+                        f"{notes + '; ' if notes else ''} Sell side of conversion from "
+                        f"{conversion_info}"
+                        f"({in_asset} in-transaction unique id: {transaction[_ID]}"
+                    ),
+                )
+            )
+
+    def _process_transfer(self, transaction: Any, intra_transaction_list: List[IntraTransaction]) -> None:
+        if transaction[_STATUS] == "failed":
+            self.__logger.info("Skipping failed transfer %s", json.dumps(transaction))
+            return None
```

Comment:
I think the pylint error is saying:
* either you remove `this return None` 
* or you add a return None outside the if statement at the end of the function (next line but one less indentation)

Since `_process_transfer()` has a `None` return type I would go for the first option.

Reply:
BTW, it's very possible pylint got updated and now it's catching more things.

Reply:
I see what happened now. I needed to pull the merge for the dividend fixes before running pylint. The static analysis error was introduced with the merge. It should be good to go now.